File: highlight_auto.js
import hljs from "highlight.js";
import fs from "fs";

// パス文字列を取得する
const path = process.argv[2];

// ファイルを非同期で読み込む
fs.readFile(path, "utf-8", (err, code) => {
  if (err) {
    console.error(err);
    return;
  }
  const result = hljs.highlightAuto(code);

  fs.writeFile(path, result.language, (err) => {
    if (err) throw err;
  });
});

File: highlight_auto.py
import guesslang
import codecs
import sys

# パス文字列を取得する
path = sys.argv[1]

# ファイルを非同期で読み込む
with codecs.open(path, "r", "utf-8") as f:
    code = f.read()

# 推測
result = guesslang.Guess().language_name(code)

# ファイルに書き込む
with codecs.open(path, "w", "utf-8") as f:
    f.write(result)